### PR TITLE
Kai Schema Fix (merge in second)

### PR DESCRIPTION
**Pull request by K-itKat20**
Previously Kai sent the schema for graphs as:
{ elements: ...., types: ...... }
This has changed to send as:
{ entities: ...., edges: ....,types: ...... }

The view graphs page has also been updated to show the UI and API URL of each graph.
#243 
#242
#238

### CODE REVIEW

**Comment by rocky341** [comment]
@t92549 
Please ignore this PR until #229 is merged in as this is built off that branch.

**Comment by rocky341** [comment]
@t92549 
Ready for review now

File: gaffer-as-a-service/gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/auth/JwtTokenUtil.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
 
     private static final long serialVersionUID = -2550185165626007488L;
 
-    public static final long JWT_TOKEN_VALIDITY = 5 * 60 * 60;
+    public static final long JWT_TOKEN_VALIDITY = 18000;
```
Comment (t92549) [inline gaffer-as-a-service/gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/auth/JwtTokenUtil.java:39]: Can this go back to 5 * 60 * 60 please?

Comment (t92549) [reply gaffer-as-a-service/gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/auth/JwtTokenUtil.java:39]: This was changed in the 229 PR but I am worried it wasn't merged into this one properly

File: gaffer-as-a-service/gaas-rest/src/main/resources/config/federated.yaml [hunk 2]
```diff
@@ -9,4 +9,4 @@ graph:
   storeProperties:
     gaffer.store.class: uk.gov.gchq.gaffer.federatedstore.FederatedStore
     gaffer.store.properties.class: uk.gov.gchq.gaffer.federatedstore.FederatedStoreProperties
-    gaffer.serialiser.json.modules: uk.gov.gchq.gaffer.sketches.serialisation.json.SketchesJsonModules
+    gaffer.serialiser.json.modules: uk.gov.gchq.gaffer.sketches.serialisation.json.SketchesJsonModules
```
Comment (t92549) [inline gaffer-as-a-service/gaas-rest/src/main/resources/config/federated.yaml:12]: Missing newline

**Review by t92549 — CHANGES_REQUESTED** [review]

File: gaffer-as-a-service/gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/auth/JwtTokenUtil.java (hunk 1 above)
Comment (rocky341) [reply gaffer-as-a-service/gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/auth/JwtTokenUtil.java:39]: Done

File: gaffer-as-a-service/gaas-rest/src/main/resources/config/federated.yaml (hunk 2 above)
Comment (rocky341) [reply gaffer-as-a-service/gaas-rest/src/main/resources/config/federated.yaml:12]: Done

File: gaas-ui/src/components/view-graphs/view-graphs-table.tsx [hunk 3]
```diff
@@ -177,11 +186,26 @@ function MainGraphTableRow(props: IGraphRow) {
                 <TableCell aria-label={"graph-status"}>
                     <StatusChip status={graph.getStatus()} />
                 </TableCell>
-                <TableCell aria-label={"graph-url"}>
-                    <a id={graph.getId()} href="placeholder" target="_blank" rel="noopener noreferrer">
+                <TableCell aria-label={"graph-url-ui"}>
+                    <a // nosemgrep
```
[the remark this answers is not in the document]
Comment (rocky341) [reply gaas-ui/src/components/view-graphs/view-graphs-table.tsx:190]: We do sanitisation on the returned URL but this isn't picked up by semgrep.

File: gaas-ui/src/domain/types-schema.ts [hunk 4]
```diff
@@ -52,15 +47,15 @@ export class TypesSchema {
...
     private validateTypes(notes: Notifications): void {
-        if (this.types.types === undefined) {
-            notes.addError("Types Schema does not contain property types");
+        if (this.types === undefined) {
+            notes.addError("Types Schema is undefined");
             return;
         }
-        if (typeof this.types.types !== "object") {
-            notes.addError(`Types is a ${typeof this.types.types} and not an object of types objects`);
+        if (typeof this.types !== "object") {
```
[the remark this answers is not in the document]
Comment (K-itKat20) [reply gaas-ui/src/domain/types-schema.ts:54]: removed and retested

File: gaas-ui/src/components/view-graphs/view-graphs-table.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (t92549) [reply gaas-ui/src/components/view-graphs/view-graphs-table.tsx:190]: Is this sanitizer enough protection?

[the remark this answers is not in the document]
Comment (rocky341) [reply gaas-ui/src/components/view-graphs/view-graphs-table.tsx:190]: Sanitisation was a semgrep suggestion so I think so yes

[the remark this answers is not in the document]
Comment (rocky341) [reply gaas-ui/src/components/view-graphs/view-graphs-table.tsx:190]: The code itself isn't from semgrep but the semgrep suggestion is to do some form of sanitisation on the URL

[the remark this answers is not in the document]
Comment (t92549) [reply gaas-ui/src/components/view-graphs/view-graphs-table.tsx:190]: Okay I see, is this sanitisation function enough? Do people use libraries? Not certain on the answer just asking

[the remark this answers is not in the document]
Comment (rocky341) [reply gaas-ui/src/components/view-graphs/view-graphs-table.tsx:190]: Well we put the URL through a regex function first and then we put it through the braintree/sanitize-url library function which is widely used

[the remark this answers is not in the document]
Comment (t92549) [reply gaas-ui/src/components/view-graphs/view-graphs-table.tsx:190]: Sorry didn't see that, looks good! 👍

**Comment by rocky341** [comment]
@t92549 
Is this ready to be merged in?

**Review by t92549 — APPROVED** [review]
Ready to merge when you want

**Comment by rocky341** [comment]
@t92549 
Yes ready to merge please

**Pull request merged**